Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/EventTag/ByEventTag.tsx

```diff
@@ -152,10 +159,14 @@ const EventTags: Application.Types.iByComponent = (props) => {
 
             <Modal Title={mode === 'Add' ? 'Add New Event Tag' : 'Edit ' + record.Name}
                 CallBack={(conf, isBtn) => {
-                    if (conf)
+                    if (conf) {
                         dispatch(EventTagSlice.DBAction({ verb: mode === 'Add' ? 'POST' : 'PATCH', record }));
-                    else if (isBtn)
-                        dispatch(EventTagSlice.DBAction({ verb: 'DELETE', record }));
+
+                    }
+                    else if (isBtn) {
+                        setMode('View');
```

Comment:
```suggestion
```
You do a setMode shortly after this if-else clause, unnecessary to do it twice

Reply:
setMode('View') is needed in the else if clause to hide the edit modal when the warning message pops up. Its also needed for the add modal that is called in the if clause. I can put setMode('View') in the if clause then